doc/nims.rst

```diff
@@ -9,31 +9,60 @@ Strictly speaking, `NimScript` is the subset of Nim that can be evaluated
 by Nim's builtin virtual machine (VM). This VM is used for Nim's compiletime
 function evaluation features.
 
-The `nim`:cmd: executable processes the ``.nims`` configuration files in
-the following directories (in this order; later files overwrite
-previous settings):
-
-1) If environment variable `XDG_CONFIG_HOME` is defined,
-   ``$XDG_CONFIG_HOME/nim/config.nims`` or
-   ``~/.config/nim/config.nims`` (POSIX) or
-   ``%APPDATA%/nim/config.nims`` (Windows). This file can be skipped
-   with the `--skipUserCfg`:option: command line option.
-2) ``$parentDir/config.nims`` where ``$parentDir`` stands for any
-   parent directory of the project file's path. These files can be
-   skipped with the `--skipParentCfg`:option: command line option.
-3) ``$projectDir/config.nims`` where ``$projectDir`` stands for the
-   project's path. This file can be skipped with the `--skipProjCfg`:option:
-   command line option.
-4) A project can also have a project specific configuration file named
-   ``$project.nims`` that resides in the same directory as
-   ``$project.nim``. This file can be skipped with the same
-   `--skipProjCfg`:option: command line option.
-
-For available procs and implementation details see `nimscript <nimscript.html>`_.
-
-
-Limitations
-===========
+
+Intro to Scripting
+==================
+
+NimScript can be used directly as a portable replacement for Bash and
+Batch files. Use `nim myscript.nims`:cmd: to run ``myscript.nims``. For example,
+installation of a program could be accomplished with this simple script:
+
+.. code-block:: nim
+
+  mode = ScriptMode.Verbose
+
+  var id = 0
+  while dirExists("foo" & $id):
+    inc id
+
+  exec "git clone https://github.com/bar/foo.git" & $id
+
+  withDir "foo" & $id & "/src":
+    exec "nim c foo"
+
+  mvFile "foo" & $id & "/src/foo".toExe, "bin/foo".toExe
+
+On Unix, you can also use the shebang `#!/usr/bin/env nim`, as long as your filename
+ends with ``.nims``:
+
+.. code-block:: nim
+
+  #!/usr/bin/env nim
+  mode = ScriptMode.Silent
+
+  echo "hello world"
+
+Use `#!/usr/bin/env -S nim --hints:off` to disable hints.
+
+
+Dry-Run and Debugging
+=====================
+
+Defining `scriptmode` on the command-line like so:
+`nim -d:scriptmode foo.nims`:cmd: ensures that it's run in `WhatIf` mode, where
+effects such as shell execution, file IO, and so on are logged to the console
+and never ran. There are additional options, see `nimscript <nimscript.html>`_
+for details.
```

Comment:
This is new functionality, felt like the old way was just such a big security issue and missed opportunity.